e2e/playwright/support/init_test.ts

```diff
@@ -5,42 +5,68 @@ import path from 'node:path';
 import fs from 'node:fs';
 
 import {test} from '@e2e-support/test_fixture';
-import {cleanUpBotDMs} from './utils';
-
-import {clearKVStoreForPlugin} from './kv';
 import {DeepPartial} from '@mattermost/types/utilities';
 import {AdminConfig} from '@mattermost/types/config';
 
+import {cleanUpBotDMs} from './utils';
+import {clearKVStoreForPlugin} from './kv';
 import {runOAuthServer} from './init_mock_oauth_server';
+import {preferencesForUser} from './user';
 
 const pluginDistPath = path.join(__dirname, '../../../dist');
 const pluginId = 'github';
 
-// # Clear plugin's KV store
-test.beforeAll(async () => {
+// # One time tasks
+test.beforeAll(async ({pw}) => {
+    const {adminClient, adminUser} = await pw.getAdminClient();
+    if (adminUser === null) {
+        throw new Error('can not get adminUser');
+    }
+
+    // Clear KV store
     await clearKVStoreForPlugin(pluginId);
+
+    // Run Mock OAuth server
     await runOAuthServer();
-});
 
-// # Upload plugin
-test.beforeEach(async ({pw}) => {
+    // Upload and enable plugin
     const files = await fs.promises.readdir(pluginDistPath);
     const bundle = files.find((fname) => fname.endsWith('.tar.gz'));
     if (!bundle) {
         throw new Error('Failed to find plugin bundle in dist folder');
     }
 
     const bundlePath = path.join(pluginDistPath, bundle);
-    const {adminClient} = await pw.getAdminClient();
-
     await adminClient.uploadPluginX(bundlePath, true);
     await adminClient.enablePlugin(pluginId);
```

Comment:
I know this might be pre-existing, but shouldnt we just upload & enable Plugin once in the `test.beforeAll` func instead of doing this before each test?

Reply:
Scratch my previous comment. I see you are already doing this. Thanks.  💯